extra_files/fv3jedi_ufs_interface_mod.F90

```diff
@@ -0,0 +1,157 @@
+! (C) Copyright 2020 NOAA
+!
+! This software is licensed under the terms of the Apache Licence Version 2.0
+! which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+
+module fv3jedi_ufs_interface_mod
+
+! iso
+use iso_c_binding
+
+! oops
+use datetime_mod
+use duration_mod
+
+! fckit
+use fckit_configuration_module,  only: fckit_configuration
+
+! fv3jedi
+use fv3jedi_geom_mod,            only: fv3jedi_geom
+use fv3jedi_geom_interface_mod,  only: fv3jedi_geom_registry
+use fv3jedi_state_mod,           only: fv3jedi_state
+use fv3jedi_state_interface_mod, only: fv3jedi_state_registry
+use fv3jedi_ufs_mod,             only: model_ufs
+
+implicit none
+private
+public :: fv3jedi_ufs_registry
+
+!> Linked list interface
+#define LISTED_TYPE model_ufs
+#include "oops/util/linkedList_i.f"
+type(registry_t) :: fv3jedi_ufs_registry
+
+! --------------------------------------------------------------------------------------------------
+
+contains
+
+! --------------------------------------------------------------------------------------------------
+
+!> Linked list implementation
+#include "oops/util/linkedList_c.f"
+
+! --------------------------------------------------------------------------------------------------
+
+subroutine c_fv3jedi_ufs_create(c_key_self, c_conf, c_key_geom) &
```

Comment:
Is there a way to pass error codes up to JEDI?  Otherwise, we end up with an abort at the model level and the application terminates.